docs/side_quests/metadata.md

```diff
@@ -0,0 +1,1039 @@
+          ch_prediction.view()
+
+      }
+    ```
+
+=== "Before"
+
+    ```groovy title="main.nf" linenums="20"
+    workflow  {
+
+        ch_samplesheet = Channel.fromPath("./data/samplesheet.csv")
+                    .splitCsv(header: true)
+                    .map { row ->
+                        [ [id:row.id, character:row.character], row.recording ]
+                    }
+                    .view()
+    }
+    ```
+
+```bash title="Identify languages"
+nextflow run main.nf
+```
+
+```console title="Identify languages"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [voluminous_mcnulty] DSL2 - revision: f9bcfebabb
+
+executor >  local (7)
+[2c/888abb] IDENTIFY_LANGUAGE (7) [100%] 7 of 7 ✔
+[[id:sampleA, character:squirrel], fr]
+[[id:sampleB, character:tux], de]
+[[id:sampleC, character:sheep], de]
+[[id:sampleD, character:turkey], en]
+[[id:sampleE, character:stegosaurus], es]
+[[id:sampleF, character:moose], fr]
+[[id:sampleG, character:turtle], it]
+```
+
+Neat, for each of our samples, we now have a language predicted. You may have noticed something else: we kept the meta data of our samples and associated it with our new piece of information. We achieved this by adding the `meta` map to the output tuple in the process:
+
+```groovy title="main.nf" linenums="12"
+output:
+      tuple val(meta), stdout
+```
+
+This is a useful way to ensure the sample-specific meta information stays connected with any new information that is generated.
+
+### 2.2 Associate the language prediction with the input file
+
+At the moment, our sample files and their language prediction are separated in two different channels: `ch_samplesheet` and `ch_predictions`. But both channels have the same meta information associated with the interesting data points. We can use the meta map to combine our channels back together.
+
+Nextflow includes many methods for combining channels, but in this case the most appropriate operator is [`join`](https://www.nextflow.io/docs/latest/operator.html#join). If you are familiar with SQL, it acts like the `JOIN` operation, where we specify the key to join on and the type of join to perform.
+
+If we check the [`join`](https://www.nextflow.io/docs/latest/operator.html#join) documentation, we can see that it joins two channels based on a defined item, by default the first item in each tuple. If you don't have the console output still available, let's run the pipeline to check our data structures. If you removed the `view()` operator from the `ch_samplesheet` add it back in:
+
+=== "After"
+
+    ```groovy title="main.nf" linenums="20" hl_lines="8 11"
+    workflow  {
+
+          ch_samplesheet = Channel.fromPath("./data/samplesheet.csv")
+                      .splitCsv(header: true)
+                      .map { row ->
+                          [ [id:row.id, character:row.character], row.recording ]
+                      }
+                      .view()
```

Comment:
```suggestion
                      .view{ "samplesheet item: " + it }
```
We should do this sort of thing if there are going to be multiple view ops active at once.

Reply:
removed becaus join was removed